### PR TITLE
Make tests use devfiles instead of stacks

### PR DESCRIPTION
**Pull request by ScrewTSW**
### What does this PR do?
This PR should fix our tests being unable to select a stack which have now been renamed to devfiles.

### What issues does this PR fix or reference?
none - adhoc

### How have you tested this PR?
prcheck

### CODE REVIEW

File: functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java [hunk 1]
```diff
@@ -108,25 +108,25 @@ public void openDialogAbout() {
   }
 
   private TestWorkspace createChe7Workspace() {
-    prepareWorkspace("che7", WORKSPACE_NAME, null);
```
Comment (ibuziuk) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java]: why workspace name is a constant and stack / devfile name is harcoded ?

File: functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java [hunk 2]
```diff
@@ -108,25 +108,25 @@ public void openDialogAbout() {
...
-  private void prepareWorkspace(String stack, String workspaceName, Double machineRam) {
+  private void prepareWorkspace(String stackID, String workspaceName, Double machineRam) {
     dashboard.waitDashboardToolbarTitle();
     LOG.info("Opening Workspaces from left menu...");
     dashboard.selectWorkspacesItemOnDashboard();
     workspaces.clickOnAddWorkspaceBtn();
     newWorkspace.waitToolbar();
     LOG.info("Selecting stack...");
-    selectStack();
+    selectStack(stackID);
```
Comment (ibuziuk) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java]: do we really want to operate with stacks in Che 7 ?

**Comment by Katka92** [comment]
Interesting - PR check 1636 started at 16:42 and was killed by signal 15 on Jenkins at 17:08. I have seen something similar when investigating periodic tests.

**Comment by Katka92** [comment]
[test]

File: functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java (hunk 2 above)
Comment (ScrewTSW) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java]: the name is misleading, It's really the element ID of a devfile.

File: functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java (hunk 1 above)
Comment (ScrewTSW) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java]: The workspace name is being generated here: https://github.com/redhat-developer/rh-che/pull/1470/files#diff-84d9e98867a01b8453d3d350179e872dL39
And we are always using the same devfile

Comment (ibuziuk) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java]: @ScrewTSW please make devfile / stack a constant also.

File: functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java (hunk 2 above)
Comment (ibuziuk) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java]: @ScrewTSW than we need a better method naming ;-)

**Comment by Katka92** [comment]
[test]

File: functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java (hunk 2 above)
Comment (ScrewTSW) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java]: I will refactor it :)

File: functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java [hunk 3]
```diff
@@ -108,25 +110,25 @@ public void openDialogAbout() {
...
-  private void prepareWorkspace(String stack, String workspaceName, Double machineRam) {
+  private void prepareWorkspace(String stackID, String workspaceName, Double machineRam) {
     dashboard.waitDashboardToolbarTitle();
     LOG.info("Opening Workspaces from left menu...");
     dashboard.selectWorkspacesItemOnDashboard();
     workspaces.clickOnAddWorkspaceBtn();
     newWorkspace.waitToolbar();
     LOG.info("Selecting stack...");
-    selectStack();
+    selectDevfile(stackID);
```
Comment (ibuziuk) [inline functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java]: @ScrewTSW why the argument is still `stackID` if in reality it is Devfile name ?

Comment (ScrewTSW) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java]: refactored

File: functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java (hunk 1 above)
Comment (ScrewTSW) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java]: done

File: functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java (hunk 3 above)
Comment (ScrewTSW) [reply functional-tests/src/test/java/com/redhat/che/functional/tests/Che7DialogAbout.java]: fixed :D also it's good that we get a few builds running, we are seeing some issues with deployments and tests getting stuck (usually problem with theia server not starting)

**Comment by ScrewTSW** [comment]
[test]

**Pull request merged**